saleor/graphql/checkout/utils.py

```diff
@@ -13,3 +16,21 @@ def prepare_insufficient_stock_checkout_validation_error(exc):
         code=exc.code.value,
         params={"variants": variant_ids},
     )
+
+
+def prevent_sync_event_circular_query(func):
+    """Prevent from resolving field in a synchronous event.
+
+    Synchronous events are not allowed to request fields that are resolved using
+    synchronous events. This prevents circular events call.
```

Comment:
I would rephrase that to be more precise:

```
"""Prevent circular dependencies in synchronous events resolvers.

Synchronous events are not allowed to request fields that are resolved using other synchronous events, which would lead to circular calls of the webhook. Using this decorator prevents such circular events resolution.

:raises CircularSubscriptionSyncEvent: When a field being resolved from a synchronous webhook's payload uses another synchronous webhook internally.
"""
```

Reply:
fixed